store/etcd/v2/etcd.go

```diff
@@ -384,11 +383,9 @@ func (s *Etcd) AtomicDelete(key string, previous *store.KVPair) (bool, error) {
 
 	delOpts := &etcd.DeleteOptions{}
 
-	if previous != nil {
-		delOpts.PrevIndex = previous.LastIndex
-		if previous.Value != nil {
-			delOpts.PrevValue = string(previous.Value)
-		}
+	delOpts.PrevIndex = previous.LastIndex
```

Comment:
Unless I'm missing something obvious, `previous` is a pointer to a `store.KVPair` thus I wouldn't move that out of the `if previous != nil` check or it might panic if someone gives a `nil` previous pair in entry. Granted, that's a much larger issue with the method signature and we should probably use an Option struct here like in other calls. But until we improve on the interface I'd leave this as is and revert this change.

Reply:
`previous` cannot be nil due to:

```go
	if previous == nil {
		return false, store.ErrPreviousNotSpecified
	}
```

Reply:
My bad, that part was hidden in the diff.